### BEstimate Analysis

##### gRNA libraries for sickle cell and cystic fibrosis
##### Used Parameters
- Editing window: 3-9
- sgRNA length: 20 nucleotides

In [2]:
import pandas, numpy, requests
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings("ignore")
data_path = "/lustre/scratch125/casm/team215mg/cd7/BEstimate/BEstimate/paper/data/"
guide_path = "/lustre/scratch125/casm/team215mg/cd7/BEstimate/BEstimate/paper/output/examples/"

/software/isg/languages/Python-3.9.10/lib/python3.9/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.26.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"


#### *HBB*

In [3]:
hbb_wt = pandas.read_csv("../output/examples/HBB_ABE_NGN_edit_df.csv")
hbb_mut = pandas.read_csv("../output/examples/HBB_ABE_NGN_mut_edit_df.csv")

In [4]:
hbb_wt[:4]

Hugo_Symbol      CRISPR_PAM_Sequence  gRNA_Target_Sequence  \
0         HBB  ACCATTGGAAAAGCAACCCCTGC  ACCATTGGAAAAGCAACCCC   
1         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   
2         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   
3         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   

             Location  Edit_Location Direction  Strand          Gene_ID  \
0  11:5229377-5229399        5229391      left      -1  ENSG00000244734   
1  11:5229372-5229394        5229391      left      -1  ENSG00000244734   
2  11:5229372-5229394        5229390      left      -1  ENSG00000244734   
3  11:5229372-5229394        5229389      left      -1  ENSG00000244734   

  Transcript_ID Exon_ID  ...  gRNA_flanking_sequences  Edit_in_Exon  \
0           NaN     NaN  ...                      NaN         False   
1           NaN     NaN  ...                      NaN         False   
2           NaN     NaN  ...                      NaN         False   
3           NaN     NaN  ...                      NaN         False   

   Edit_in_CDS   GC%  # Edits/guide  Poly_T  mutation_on_guide  \
0        False  50.0              1   False              False   
1        False  55.0              4   False              False   
2        False  55.0              4   False              False   
3        False  55.0              4   False              False   

   guide_change_mutation  mutation_on_window  mutation_on_PAM  
0                  False               False            False  
1                  False               False            False  
2                  False               False            False  
3                  False               False            False  

[4 rows x 21 columns]

In [5]:
hbb_mut[:4]

Hugo_Symbol      CRISPR_PAM_Sequence  gRNA_Target_Sequence  \
0         HBB  ACCATTGGAAAAGCAACCCCTGC  ACCATTGGAAAAGCAACCCC   
1         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   
2         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   
3         HBB  TGGAAAAGCAACCCCTGCCTTGA  TGGAAAAGCAACCCCTGCCT   

             Location  Edit_Location Direction  Strand          Gene_ID  \
0  11:5229377-5229399        5229391      left      -1  ENSG00000244734   
1  11:5229372-5229394        5229391      left      -1  ENSG00000244734   
2  11:5229372-5229394        5229390      left      -1  ENSG00000244734   
3  11:5229372-5229394        5229389      left      -1  ENSG00000244734   

  Transcript_ID Exon_ID  ...  gRNA_flanking_sequences  Edit_in_Exon  \
0           NaN     NaN  ...                      NaN         False   
1           NaN     NaN  ...                      NaN         False   
2           NaN     NaN  ...                      NaN         False   
3           NaN     NaN  ...                      NaN         False   

   Edit_in_CDS   GC%  # Edits/guide  Poly_T  mutation_on_guide  \
0        False  50.0              1   False              False   
1        False  55.0              4   False              False   
2        False  55.0              4   False              False   
3        False  55.0              4   False              False   

   guide_change_mutation  mutation_on_window  mutation_on_PAM  
0                  False               False            False  
1                  False               False            False  
2                  False               False            False  
3                  False               False            False  

[4 rows x 21 columns]

In [6]:
hbb_mut[hbb_mut.guide_change_mutation]

Hugo_Symbol      CRISPR_PAM_Sequence  gRNA_Target_Sequence  \
2315         HBB  ACTTCTCCACAGGAGTCAGATGC  ACTTCTCCACAGGAGTCAGA   

                Location  Edit_Location Direction  Strand          Gene_ID  \
2315  11:5226994-5227016        5227002     right      -1  ENSG00000244734   

        Transcript_ID          Exon_ID  ...  gRNA_flanking_sequences  \
2315  ENST00000335295  ENSE00001829867  ...                      NaN   

      Edit_in_Exon  Edit_in_CDS   GC%  # Edits/guide  Poly_T  \
2315          True         True  50.0              1   False   

      mutation_on_guide  guide_change_mutation  mutation_on_window  \
2315               True                   True                True   

      mutation_on_PAM  
2315            False  

[1 rows x 21 columns]

#### David Liu paper

- WT codon is *GAG* and the mutant one is *GTG*
- Mutation is on 5227002 and mutant sequence between 5227001-5227003 is CAC --> ABE --> CGC (+1 strand)

GCG --> Ala --> naturally occurring, non-sickling variant hemoglobin "Makassar" (HbG) [https://doi.org/10.1182/blood-2020-139016]

In [7]:
uniprot_server = "https://www.ebi.ac.uk/proteins/api/"
uniprot_api = "proteins?offset=0&size=-1&accession=D9YZU5" 
api_request = requests.get(uniprot_server + uniprot_api, headers={"Accept": "application/json"})
hbb_seq = None
for i in api_request.json():
    hbb_seq = i["sequence"]["sequence"]

hbb_seq

'MVHLTPEEKSAVTALWGKVNVDEVGGEALGRLLVVYPWTQRFFESFGDLSTPDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFATLSELHCDKLHVDPENFRLLGNVLVCVLAHHFGKEFTPPVQAAYQKVVAGVANALAHKYH'

In [20]:
server = "https://rest.ensembl.org"
hugo_ensembl = "/sequence/region/human/11:5226994-5227016?content-type=text/plain"
gene_request = requests.get(server + hugo_ensembl, headers={"Content-Type": "application/json"})
gene_request.json()

{'molecule': 'dna',
 'query': '11:5226994-5227016',
 'seq': 'ACTTCTCCTCAGGAGTCAGATGC',
 'id': 'chromosome:GRCh38:11:5226994:5227016:1'}